platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java

```diff
@@ -97,29 +98,59 @@ public void onReceived(ObservedPayment payment) {
       return;
     }
 
+    // Build Stellar Transaction object
+    StellarTransaction stellarTransaction =
+        StellarTransaction.builder()
+            .id(payment.getTransactionHash())
+            .memo(txn.getStellarMemo())
+            .memoType(txn.getStellarMemoType())
+            .createdAt(parsePaymentTime(payment.getCreatedAt()))
+            .envelope(payment.getTransactionEnvelope())
+            .payments(
+                List.of(
+                    StellarPayment.builder()
+                        .id(payment.getId())
+                        .paymentType(
+                            payment.getType() == ObservedPayment.Type.PAYMENT
+                                ? StellarPayment.Type.PAYMENT
+                                : StellarPayment.Type.PATH_PAYMENT)
+                        .sourceAccount(payment.getFrom())
+                        .destinationAccount(payment.getTo())
+                        .amount(new Amount(payment.getAmount(), payment.getAssetName()))
+                        .build()))
+            .build();
+
+    TransactionEvent.Status oldStatus = TransactionEvent.Status.from(txn.getStatus());
+    TransactionEvent.Status newStatus;
+    String message;
+
     // Check if the payment contains the expected amount (or greater)
     BigDecimal expectedAmount = decimal(txn.getAmountIn());
     BigDecimal gotAmount = decimal(payment.getAmount());
     if (gotAmount.compareTo(expectedAmount) < 0) {
-      Log.warn(
+      updateTransactionStatusTo(ERROR, txn, payment, stellarTransaction);
+
+      newStatus = TransactionEvent.Status.ERROR;
+      message =
           String.format(
-              "Payment amount %s is smaller than the expected amount %s",
-              payment.getAmount(), txn.getAmountIn()));
-      updateTransactionStatusTo(ERROR, txn, payment);
-      return;
+              "The incoming payment amount was insufficient! Expected: \"%s\", Received: \"%s\"",
+              formatAmount(expectedAmount), formatAmount(gotAmount));
+      Log.warn(message);
+    } else {
+      updateTransactionStatusTo(PENDING_RECEIVER, txn, payment, stellarTransaction);
+
+      newStatus = TransactionEvent.Status.PENDING_RECEIVER;
+      message = "Incoming payment for SEP-31 transaction";
+      Log.info(message);
     }
```

Comment:
So we're saying that if the amount sent is less than expected, the transaction's status will remain in `pending_sender`? Polaris updates the transaction status to `pending_receiver` regardless of the amount sent, so we should probably discuss what makes sense here so we can be consistent.

My thought is that the anchor needs to decide how to proceed if the amount sent isn't what was expected, so we should change the status to `pending_receiver`, but I could be convinced otherwise.

Reply:
> So we're saying that if the amount sent is less than expected, the transaction's status will remain in `pending_sender`?

No, if the amount is smaller than expected, the status changes to `ERROR` and an event is triggered. This indicates that the receiving anchor should handle this error on their side and decide what to do later.

I'm happy to discuss this further, I don't think the SEP enforces a specific behavior here but as you said, it's better to be consistent.

Reply:
Hmm, I'm not sure `error` makes sense here. If the anchor wants to continue facilitating the transaction, they'd have to update the status back to `pending_receiver` or `pending_sender`, and the Sending Anchor may stop watching for updates after they see the `error` status.

Thats why I think using `pending_receiver` makes sense. If the anchor wants to change it to `error` it can, but they could also continue facilitating the transaction without changing the status.

Reply:
So your rationale is that any received transaction (with the expected memo) would trigger a pending_receiver, and the receiver can handle that however they want. And they should validate the amounts are correct on their side as well and maybe mark things as `error` or `refunded` if they want to?

I think that kinda works, because the [statuses section from SEP-31](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0031.md#get-transaction) states:
> `pending_receiver` -- payment is being processed by the Receiving Anchor.

This means if the payment amount is smaller than expected, the payment will be processed and marked as error or refunded, depending on the Anchor policy. 

Alright, I'll proceed with that.